### PR TITLE
Support nested keys as alternative to hash_key for fields

### PR DESCRIPTION
**Pull request by joelzwarrington**
closes https://github.com/rmosolgo/graphql-ruby/issues/3855

### CODE REVIEW

**Comment by joelzwarrington** [comment]
@rmosolgo I have something but am not entirely pleased with how I'm setting `method_sym` and overloading the shit out of it.

Also I'm explicitly not using `.to_sym` on each of the keys as it causes issues when not using an indifferent hash.

File: lib/graphql/schema/field.rb [hunk 1]
```diff
@@ -236,26 +237,26 @@ def initialize(type: nil, name: nil, owner: nil, null: true, field: nil, functio
...
-          if hash_key
-            raise ArgumentError, "Provide `hash_key:` _or_ `resolver_method:`, not both. (called with: `hash_key: #{hash_key.inspect}, resolver_method: #{resolver_method.inspect}`)"
+          if hash_key || dig
+            raise ArgumentError, "Provide `hash_key:`, `dig:`, _or_ `resolver_method:`, not multiple. (called with: `hash_key: #{hash_key.inspect}, dig: #{dig.inspect}, resolver_method: #{resolver_method.inspect}`)"
           end
         end
 
         # TODO: I think non-string/symbol hash keys are wrongly normalized (eg `1` will not work)
-        method_name = method || hash_key || name_s
+        method_name = method || hash_key || dig || name_s
```
Comment (rmosolgo) [inline lib/graphql/schema/field.rb:255]: I appreciate you're integrating with the existing code, but given that the previous code here is not ... _exemplary_ ... (😅 ), how about keeping `dig` _separate_ from the existing (confusing) attributes? I'll leave some suggestions to that effect...

```suggestion
        method_name = method || hash_key || name_s
        @dig_keys = dig
```

File: lib/graphql/schema/field.rb [hunk 2]
```diff
@@ -236,26 +237,26 @@ def initialize(type: nil, name: nil, owner: nil, null: true, field: nil, functio
...
         resolver_method ||= name_s.to_sym
 
         @method_str = -method_name.to_s
-        @method_sym = method_name.to_sym
+        @method_sym = method_name.is_a?(Array) ? method_name : method_name.to_sym
```
Comment (rmosolgo) [inline lib/graphql/schema/field.rb:259]:
```suggestion
        @method_sym =  method_name.to_sym
```

**Review by rmosolgo — COMMENTED** [review]
The API and tests look good to me -- I just have one suggestion under the hood, let me know what you think of it.

**Comment by joelzwarrington** [comment]
> The API and tests look good to me -- I just have one suggestion under the hood, let me know what you think of it.

Thanks @rmosolgo, I can make some changes based on your feedback.

Do you have anything planned to revisit some of the underlying API stuff? I'd be open to helping out and doing some refactoring if needed 😄

**Comment by rmosolgo** [comment]
>  anything planned

No, nothing planned! I'd be happy to discuss in an issue or review a PR if you have something in mind, though :+1:

**Comment by rmosolgo** [comment]
I'll work up a fix for rails master elsewhere. Thanks for this improvement!

**Pull request merged**